spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Unsqueeze.scala

```diff
@@ -32,46 +32,51 @@ import scala.reflect.ClassTag
 
 @SerialVersionUID(- 5180889605872472241L)
 class Unsqueeze[T: ClassTag](
-  val pos: Int,
+  val pos: Array[Int],
   var numInputDims: Int = Int.MinValue
 )(implicit ev: TensorNumeric[T]) extends AbstractModule[Tensor[_], Tensor[_], T]  {
 
   def setNumInputDims(numInputDims: Int): Unit = {
     this.numInputDims = numInputDims
   }
 
-  private def getActualPosition(input: Tensor[_]) : Int = {
-    val dim = if (pos <= 0) {
-      input.dim() + pos + 1
-    } else {
-      pos
-    }
-
-    // get valid dimension offset for batchMode (if any)
-    val inputDim = input.dim() // data batch dim
-    numInputDims = if (numInputDims != Int.MinValue) numInputDims else inputDim // feature map dim
-    val offsetDim = inputDim - numInputDims
-    require(offsetDim >= 0, "input feature map dim (numInputDims) must be <= input:dim()," +
-      s" input feature map dim ${numInputDims}, inputdim ${inputDim}")
-
-    // the actual position; clearer error message for batchMode (if any)
-    val actualPos = dim + offsetDim
-    require(actualPos >= 1 && actualPos <= (inputDim + 1), s"Invalid position: $pos. " +
-      s"input:dim() is $input, input feature map dim (numInputDims) is $numInputDims.")
-
-    actualPos
+  private def getActualPosition(input: Tensor[_]) : Array[Int] = {
+    pos.foreach(pos => {
+      val dim = if (pos <= 0) {
```

Comment:
Please try to simplify the logic here.
Suggest get rid of If Else statement.
The Require statement enforces pos > 0.

Reply:
the original function takes 0 as input.dim() + pos + 1. Are you suggesting we take this out and only take numbers >0 ?

Reply:
How are you going to treat the input if it is 2, do you know it is starting from 0 or 1?

Reply:
the code assume the input start from 1, and there is a require to make sure the pos is no larger than the input size:
require(actualPos >= 1 && actualPos <= (inputDim + 1), s"Invalid position: $pos. " +
        s"input:dim() is $input, input feature map dim (numInputDims) is $numInputDims.")

Reply:
I dont understand.
The If statement here guarantee pos <= 0, why there is a require statement after it?